<a href="https://www.kaggle.com/code/nanduvardhanreddy/king-kohli-eda?scriptVersionId=115278142" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv('../input/virat-kohlis-71-centuries/71 Centuries of Virat Kohli.csv')
df.head()

Score Out/Not Out       Against  Batting Order  Inn.  Strike Rate  \
0    116         Out     Australia              6     2          NaN   
1    103         Out   New Zealand              5     2          NaN   
2    103         Out       England              5     2          NaN   
3    107         Out     Australia              5     2          NaN   
4    119         Out  South Africa              4     1          NaN   

                                  Venue        Column1   H/A        Date  \
0                         Adelaide Oval       Adelaide  Away  24-01-2012   
1                M. Chinnaswamy Stadium      Bangalore  Home  31-08-2012   
2  Vidarbha Cricket Association Stadium         Nagpur  Home  13-12-2012   
3             M. A. Chidambaram Stadium        Chennai  Home  22-02-2013   
4                     Wanderers Stadium   Johannesburg  Away  18-12-2013   

  Result Format Man of the Match Captain  Unnamed: 14  
0   Lost   Test               No      No          NaN  
1    Won   Test              Yes      No          NaN  
2  Drawn   Test               No      No          NaN  
3    Won   Test               No      No          NaN  
4  Drawn   Test               No      No          NaN

In [3]:
df['Unnamed: 14'].isnull().sum()

71

let's remove unnamed :14, Strike rate and date because all are null values and rename Column1 as Match Place 

In [4]:
df.drop('Unnamed: 14',inplace=True,axis=1)
df.rename(columns={'Column1':'Match Place'},inplace=True)
df.drop('Strike Rate',inplace=True,axis=1)
df.drop('Date',inplace=True,axis=1)


In [5]:
df.info() # FINALLY NO NULL VALUES

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Score             71 non-null     int64 
 1   Out/Not Out       71 non-null     object
 2   Against           71 non-null     object
 3   Batting Order     71 non-null     int64 
 4   Inn.              71 non-null     int64 
 5   Venue             71 non-null     object
 6   Match Place       71 non-null     object
 7   H/A               71 non-null     object
 8   Result            71 non-null     object
 9   Format            71 non-null     object
 10  Man of the Match  71 non-null     object
 11  Captain           71 non-null     object
dtypes: int64(3), object(9)
memory usage: 6.8+ KB


# UNIVARIATE ANALYSIS

# 1.Highest Score

In [6]:
df.sort_values(by='Score',ascending=False).head(1)

Score Out/Not Out       Against  Batting Order  Inn.  \
25    254     Not Out  South Africa              4     1   

                                      Venue Match Place   H/A Result Format  \
25  Maharashtra Cricket Association Stadium        Pune  Home    Won   Test   

   Man of the Match Captain  
25              Yes     Yes

**HIGHEST SCORE BY KOHLI IS 254 AND IT'S AGAINST SOUTH AFRICA IN A TEST MATCH AS A CAPTAIN AND INDIA WON THE MATCH**

# 1.1 Most Centuries

In [7]:
df['Venue'].value_counts().head(5)

Adelaide Oval                           5
Sher-e-Bangla Cricket Stadium           4
Vidarbha Cricket Association Stadium    3
Queen's Park Oval                       3
R. Premadasa Stadium                    3
Name: Venue, dtype: int64

In [8]:
fig=px.bar(x=df['Venue'].value_counts().head(5).index,
       y=df['Venue'].value_counts().head(5).values,
       text=df['Venue'].value_counts().head(5).values,
       color=df['Venue'].value_counts().head(5).index,
       title='TOP 5 MOST CENTURIES IN STADIUM',
       template='plotly_dark')
fig.update_layout(xaxis_title='STADIUM',yaxis_title='NO.OF CENTURIES')


**OUT OF 71 CENTURIES MAXIMUM CENTURIES ARE IN ADELAIDE OVAL STADIUM**

# 1.2 Most Centuries against

In [9]:
fig=px.bar(x=df['Against'].value_counts().index,
       y=df['Against'].value_counts().values,
       text=df['Against'].value_counts().values,
       color=df['Against'].value_counts().index,
       title='MOST CENTURIES AGAINST',
       template='plotly_dark')
fig.update_layout(xaxis_title='COUNTRY',yaxis_title='NO.OF CENTURIES')


**HE HAD SCORE MORE CENTURIES AGAINST AUSTRALIA**

In [10]:
df['Out/Not Out'].value_counts()

Out        51
Not Out    20
Name: Out/Not Out, dtype: int64

# 1.3 OUT VS NOT OUT

In [11]:
fig=px.pie(names=df['Out/Not Out'].value_counts().index,
           values=df['Out/Not Out'].value_counts().values,
           title='OUT VS NOT OUT',
            template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

In [12]:
fig=px.pie(names=df['Batting Order'].value_counts().index,
           values=df['Batting Order'].value_counts().values,
           title='Batting Order',
            template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

**HALF THE CENTURIES ARE SCORED IN PLAYING AT 3RD ORDER**

In [13]:
fig=px.pie(names=df['Man of the Match'].value_counts().index,
           values=df['Man of the Match'].value_counts().values,
           title='Man of the Match',
            template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

**MORE THAN HALF TIMES KOHLI WON THE MAN OF THE MATCH**

# CENTURIES AS A CAPTAIN

In [14]:
fig=px.pie(names=df['Captain'].value_counts().index,
           values=df['Captain'].value_counts().values,
           title='Captain',
            template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

**HE HAD SCORED MORE CENTURIES AS A CAPTAIN THAN HE IS NOT**

# 2. BIVARIATE ANALYSIS

# 2.1 INDIA WON

In [15]:
df.groupby(by='Out/Not Out')['Result'].value_counts()

Out/Not Out  Result    
Not Out      Won           17
             Drawn          2
             Tied           1
Out          Won           31
             Lost          13
             Drawn          5
             Lost (D/L)     1
             Won (D/L)      1
Name: Result, dtype: int64

**INDIA ONLY LOST 2 MATCHES OUT OF 20 WHEN KOHLI WAS NOT OUT. SO, THERE IS THE HIGH PROBABILTY OF INDIA WINNING IF KOHLI WAS NOT OUT**

In [16]:
df.groupby(by='Result')['Format'].value_counts()

Result      Format
Drawn       Test       7
Lost        Test       7
            ODI        6
Lost (D/L)  ODI        1
Tied        ODI        1
Won         ODI       34
            Test      13
            T20I       1
Won (D/L)   ODI        1
Name: Format, dtype: int64